src/MongoDB.Bson/Serialization/Conventions/ImmutableTypeClassMapConvention.cs

```diff
@@ -25,28 +25,34 @@ namespace MongoDB.Bson.Serialization.Conventions
     /// </summary>
     public class ImmutableTypeClassMapConvention : ConventionBase, IClassMapConvention
     {
+        private readonly BindingFlags _constructorBindingFlags;
+        private readonly BindingFlags _propertyBindingFlags;
+
+        /// <inheritdoc />
+        public ImmutableTypeClassMapConvention()
+        {
+            _constructorBindingFlags = BindingFlags.Public | BindingFlags.Instance | BindingFlags.NonPublic;
```

Comment:
allow considering protected constructors

Reply:
Normally we put visibility flags first and then Instance or Static (same as on C# declaration line).

    BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance

Reply:
We need to ignore private constructors somehow.

Looks like they have to be filtered out some other way after calling GetConstructors (by checking IsPrivate property?).

Reply:
Done